Syntax lookup: Deprecated decorator (#349)
rescript-association/rescript-lang.org: misc_docs/syntax/decorator_module_deprecated.mdx

---
id: "module-deprecated-decorator"
keywords: ["deprecated", "decorator"]
name: "@@deprecated"
summary: "This is the `@@deprecation` decorator."
category: "decorators"
---

The `@@deprecated` decorator is used to add a deprecation note to the file-level of a module. The compiler and editor tooling will yield a warning whenever a deprecated file module is being used.

> Use the `@deprecated` decorator to deprecate specific types, values and submodules.

Reviewer: ```suggestion
For more fine-grained control, use the `@deprecated` decorator to add deprecation warnings to specific types, values and submodules.
```

---
id: "module-deprecated-decorator"
keywords: ["deprecated", "decorator"]
name: "@@deprecated"

> summary: "This is the `module deprecation` decorator."

Reviewer: ```suggestion
summary: "This is the `@@deprecation` decorator."
```

category: "decorators"
---

> The `@@deprecated` decorator is used to indicate that a whole module is deprecated. This may be used by ReScript tooling to highlight deprecated code during development.

Reviewer: ```suggestion
The `@@deprecated` decorator is used to indicate that a whole module is deprecated. This may be used by ReScript tooling to highlight deprecated code during development.

The `@@deprecated` decorator is used to add a deprecation notes to the file-level of a module. The compiler and editor tooling will yield a warning whenever a deprecated file module is being used. This is useful to gracefully deprecate file modules and give your users a hint for future migration paths.
```

> See the `@deprecated` decorator to deprecate expressions.

Reviewer: ```suggestion
Use the `@deprecated` decorator to deprecate specific types and values.
```
